ios/engine/KMEI/KeymanEngine/Classes/KeyboardRepository/APIKeyboardRepository.swift

```diff
@@ -75,7 +76,32 @@ public class APIKeyboardRepository: KeyboardRepository {
     }
 
     options = result.options
-    languages = Dictionary(uniqueKeysWithValues: result.languages.map { ($0.id, $0) })
+
+    // Detect duplicate ids - this scenario indicates an infrastructure problem.
+    var dictionary = Dictionary<String, Language>()
+    var duplicateDetected = false
+    result.languages.forEach { language in
+      // Does an entry for this ID already exist?  Trouble if so.
+      if let _ = dictionary[language.id] {
+        duplicateDetected = true
+        return
+      }
+
+      dictionary.updateValue(language, forKey: language.id)
+    }
+
+    guard !duplicateDetected else {
+      // This indicates an issue in our keyboard API infrastructure.
+      // Don't let it crash the app, but definitely make note of it
+      // as something to be fixed.
+      log.error("Corrupt data detected in API returns: duplicate language codes")
```

Comment:
To help with future troubleshooting, can this include the duplicated language code?

Reply:
Might have to be "code**s**"... it should be possible to track an array for this.  Assuming there's only one may be a bit naive.

Reply:
Good point. If  the API goes really pear-shaped, maybe just a handful of codes will suffice to see what's going on.